test/config-luatest/config_test.lua

```diff
@@ -10,10 +10,14 @@ local json = require('json')
 local g = t.group()
 
 g.before_all(function()
+    g.old_box_cfg = box.cfg
+    box.cfg = function() end
     treegen.init(g)
 end)
 
 g.after_all(function()
+    box.cfg = g.old_box_cfg
+    box.cfg = g.old_box_cfg
     treegen.clean(g)
 end)
```

Comment:
Nit: Looks like it is squashed into a wrong commit, because this is the only change in the test file.

Reply:
Nah, it's not. Everything in this test works inside luatest instance, `box.cfg` of which is a table, but it's not actually configured. So, when we're trying to access spaces to get saved names, the test fails.

Reply:
Got it, the problem is in the `test_configdata` test case. The `configdata` module now leans on `box.cfg`.

OK. I can accept this workaround, but I would prefer to run all the box dependent code in a separate process. For example:

```diff
diff --git a/test/config-luatest/config_test.lua b/test/config-luatest/config_test.lua
index 2253257532..fafa3d5771 100644
--- a/test/config-luatest/config_test.lua
+++ b/test/config-luatest/config_test.lua
@@ -1,7 +1,5 @@
 local t = require('luatest')
 local server = require('test.luatest_helpers.server')
-local cluster_config = require('internal.config.cluster_config')
-local configdata = require('internal.config.configdata')
 local helpers = require('test.config-luatest.helpers')
 local treegen = require('test.treegen')
 local justrun = require('test.justrun')
@@ -10,14 +8,10 @@ local json = require('json')
 local g = t.group()
 
 g.before_all(function()
-    g.old_box_cfg = box.cfg
-    box.cfg = function() end
     treegen.init(g)
 end)
 
 g.after_all(function()
-    box.cfg = g.old_box_cfg
-    box.cfg = g.old_box_cfg
     treegen.clean(g)
 end)
 
@@ -27,7 +21,20 @@ g.after_each(function()
     end
 end)
 
-g.test_configdata = function()
+local function verify_configdata()
+    local json = require('json')
+    local t = require('luatest')
+    local configdata = require('internal.config.configdata')
+    local cluster_config = require('internal.config.cluster_config')
+
+    local saved_assert_equals = t.assert_equals
+    t.assert_equals = function(...)
+        local ok, err = pcall(saved_assert_equals, ...)
+        if not ok then
+            error(json.encode(err), 2)
+        end
+    end
+
     local cconfig = {
         credentials = {
             users = {
@@ -151,6 +158,20 @@ g.test_configdata = function()
     t.assert_equals(data:peers(), {'instance-001', 'instance-002'})
 end
 
+g.test_configdata = function()
+    local dir = treegen.prepare_directory(g, {}, {})
+    local script = string.dump(verify_configdata)
+    treegen.write_script(dir, 'main.lua', script)
+
+    local opts = {nojson = true, stderr = true}
+    local res = justrun.tarantool(dir, {}, {'main.lua'}, opts)
+    t.assert_equals(res, {
+        exit_code = 0,
+        stdout = '',
+        stderr = '',
+    })
+end
+
 g.test_config_general = function()
     local dir = treegen.prepare_directory(g, {}, {})
     local script = [[
```

Reply:
(Feel free to resolve the discussion with any solution you prefer.)

Reply:
Sure, done. Thank you for the diff!

P.S. Also updated alert message for the instance, uuid of which is not known